### DISCUSSION TITLE
Is there an ui.update_input_file and ui.update_download_button?

### DISCUSSION THREAD

**Discussion by gernophil** [posit-dev/py-shiny#1783] [Q&A]
I already asked this at SO, but I hope people here will know better :).

I have a Shiny for Python app that uses `ui.input_file()` and `ui.download_button()`. For most of the UI Inputs there exists a corresponding `ui.update_...()` with that you can activate/deactivate a button (`ui.update_action_button(id="some_id", disabled=False)` or simply undo and made input (`ui.update_text("some_other_id", value='')`).

Is there a way to clear the selected file in an `ui.input_file()` input and to enable disable an `ui.download_button`?

**Comment by gadenbuie** [comment]
We have an open issue for this: https://github.com/posit-dev/py-shiny/issues/1583. We did a little preliminary discovery work and I think we've found a route to make it possible, but it might take us a little to get to it. You can subscribe to that issue to make sure you get updates.

**Reply by gernophil** [reply]
A great, thanks for the info. Must have overlooked this :).
